lib/logstash/codecs/kms.rb

```diff
@@ -0,0 +1,140 @@
+# encoding: utf-8
+require "logstash/codecs/base"
+require "logstash/namespace"
+require "logstash-codec-kms_jars"
+
+# A codec to encrypt/decrypt messages using AWS KMS
+#
+# This plugin uses the AWS Encryption SDK vended by Amazon.
+# See http://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/introduction.html for the details.
+#
+# To use this plugin, you *must*:
+#
+#  * Have an AWS account.
+#  * Setup a KMS key to use.
+#  * Create an identify that has access to decrypt using the kms key you created.
+#
+# See https://docs.aws.amazon.com/kms/latest/developerguide/create-keys.html
+# for details on creating keys.
+#
+# See http://docs.aws.amazon.com/kms/latest/developerguide/control-access-overview.html#managing-access
+# for details on seting up permissions for AWS KMS.
+#
+class LogStash::Codecs::Kms < LogStash::Codecs::Base
+
+  config_name "kms"
+
+  # The codec use after the message is decrypted (if this is used with an input)
+  # or the codec to use before the message is encrypted (if this is used with an output)
+  # [source,ruby]
+  #     stdin {
+  #         codec => kms {
+  #             codec => "json"
+  #         }
+  #     }
+  #
+  config :codec, :validate => :codec, :default => "plain"
+
+  # A list of KMS key ids. If more than one key is provided, the encrypted payload will
+  # contain a copy of the data key encrypted with each of the keys specified here.
+  # At minimun, you should provide a single key id to use for KMS encryption.
+  config :key_ids, :validate => :string, :list => true, :required => true
```

Comment:
With the latter I also get to specify that I expect each element in the array to be a string.